### ISSUE TITLE
Cross join field aliasing?

### ISSUE THREAD

**Issue by rudolfdobias** [aspnet/EntityFramework#8451]
Hello,
I suspect EF is misbehaving when it comes to cross join on the same table. The fileds in select statement aren't aliased and the duplicated column names are merged on the output.

### Steps to reproduce

I've following expression:
(just a cartesian product of something)

```
                    var data = (
                    from a in _db.CurrencyRatios
                    from b in _db.CurrencyRatios
                    where a.CreatedAt >= ...
                    where b.CreatedAt >= ...
                    select new
                    {
                        From = a.From,
                        To = b.From,
                        Ratio = a.Ratio / b.Ratio
                    }
                    ).ToList();
```
note: it doesn't matter if the selecting object is anonymous or not

Translated SQL:

```
      SELECT "a"."From", "b"."From", "a"."Ratio" / "b"."Ratio"
      FROM "travel"."CurrencyRatios" AS "a"
      CROSS JOIN "travel"."CurrencyRatios" AS "b"
      WHERE ("a"."CreatedAt" >= @__lastDate_0) AND ("b"."CreatedAt" >= @__lastDate_1)
```

The result: 
Collection of one object like `[{"USD", "USD", 1}]`

The problem:
The table contains about 100 rows so I'm expecting about 10k of results. IMO it happens because the `From` fields aren't aliased.
Is this a bug/feature or it's happening somewhere else (npgsql, linq)? Or is there some other way?

Thanks

### Further technical details
EF Core version: 1.1.0
Database Provider: (Postgres - Npgsql 1.1.0)
Operating system: OSX 10.12.1
IDE: Rider EAP 17

**Comment by smitpatel** [comment]
#8242 will introduce aliasing. So generated query will be different.

Though we should investigate how it got mixed up without aliasing. It should not have happened.

**Comment by maumar** [comment]
Verified that this works in current bits. We produce the following sql:

```sql
SELECT [a].[From] AS [From0], [b].[From] AS [To], [a].[Ratio] / [b].[Ratio] AS [Ratio]
FROM [CurrencyRatios] AS [a]
CROSS JOIN [CurrencyRatios] AS [b]
WHERE ([a].[CreatedAt] >= @__date1_0) AND ([b].[CreatedAt] >= @__date2_1)
```

**Comment by rudolfdobias** [comment]
Thanks for investigation and fix. 
Coincidentally, about one hour ago I found  that the aliasing wasn't the main culprit of my issue - it was the CreatedAt condition.
The *@__lastDate* argument is a DateTime drawn from DB in previous query (like _select createdAt from .. order by createdAt desc limit 1)_  In the dumped query it looks like 

    WHERE "CreatedAt" = '2018-15-15T23:45:17'

But in a fact the argument contained the ticks included, like '2018-15-15T23:45:17.0124325' but it wasn't displayed in the console ... And since every record in the DB has slightly more tick than the previous .. you can think the rest.

But the fixed aliasing will be definitely a valuable addition to the framework anyway.
